examples/hpc/dla-driver/src/tensor3.rs

```diff
@@ -0,0 +1,248 @@
+#![no_std]
+#![no_main]
+
+use alloc::vec::*;
+use headsail_bsp::ufmt::uDisplay;
+use headsail_bsp::{sprint, sprintln};
+use ndarray::{Array, Array3};
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Order3 {
+    CHW,
+    CWH,
+    HWC,
+    HCW,
+    WHC,
+    WCH,
+}
+
+impl From<Order3> for [usize; 3] {
+    fn from(order: Order3) -> Self {
+        match order {
+            Order3::CHW => [0, 1, 2],
+            Order3::CWH => [0, 2, 1],
+            Order3::HWC => [1, 2, 0],
+            Order3::HCW => [1, 0, 2],
+            Order3::WHC => [2, 1, 0],
+            Order3::WCH => [2, 0, 1],
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Tensor3<T> {
+    data: Array3<T>,
+    pub channels: usize,
+    pub height: usize,
+    pub width: usize,
```

Comment:
`data` seems to already contain the information for `channels`, `height`, `width`. These three fields are now duplicated and need to be synchronized everywhere which isn't great.

Reply:
In case you need easy access to the symbolic `channels`, `height`, `width` values (which are `order`-dependent), replace these three fields with three functions with the same name, which are able to resolve the correct dimension based on the `order` field.

Reply:
Added functions to access channels, height, and width.